crowbar_framework/lib/crowbar/deployment_queue.rb

```diff
@@ -17,25 +17,23 @@ module Crowbar
   class DeploymentQueue
     include CrowbarPacemakerProxy
 
-    attr_reader :logger, :proposal_queue
+    attr_reader :logger
 
     def initialize(logger: Rails.logger)
-      @logger         = logger
-      @proposal_queue = ::ProposalQueue.ordered.first
+      @logger = logger
     end
 
     # Receives proposal info (name, barclamp), list of nodes (elements), on which the proposal
     # should be applied, and list of dependencies - a list of {barclamp, name/inst} hashes.
     # It adds them to the queue, if possible.
     def queue_proposal(bc, inst, elements, element_order, deps)
-      logger.debug("queue proposal: enter #{inst} #{bc}")
+      logger.debug("queue proposal: enter for #{bc}:#{inst}")
       delay = []
       pre_cached_nodes = {}
       begin
         lock = acquire_lock("queue")
-        if @proposal_queue.barclamp == bc && @proposal_queue.name == inst
-          preexisting_queued_item = @proposal_queue.properties
-        end unless @proposal_queue.nil?
+
+        queued_proposal = ProposalQueue.find_by(barclamp: bc, name: inst)
```

Comment:
For one proposal in a barclamp, no.

For one barclamp, yes: two different proposals.